### PR TITLE
[Heartbeat] Experimental Script monitors

### PR DESCRIPTION
**Pull request by andrewvc**
Experiment

### CODE REVIEW

File: heartbeat/monitors/active/synthetic/script.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		//defer os.RemoveAll(dataDir)
+
+		err = ioutil.WriteFile(scriptPath, []byte(config.Script), 0644)
+		if err != nil {
+			return fmt.Errorf("could not write script file %s: %w", scriptPath, err)
+		}
+
+		logp.Info("Start script job")
+		cmd := exec.Command(
+			"node",
```
Comment (vigneshshanmugam) [inline heartbeat/monitors/active/synthetic/script.go:50]: What do you think about shipping a bin file from our package? and hearbeat can invoke our node script using shell? 

```
./elastic-synthetic  /path/to/script.js

```

Comment (vigneshshanmugam) [reply heartbeat/monitors/active/synthetic/script.go:50]: There is also other cool way to do this, Hearbeat can pipe this code and Node.js can read it from stdin. Advantage is multiple sources can be passed, we can also align on a separator that would provide distinction to journeys.

Comment (andrewvc) [reply heartbeat/monitors/active/synthetic/script.go:50]: ++ to this, already done

File: heartbeat/monitors/active/script/script.go [hunk 2]
```diff
@@ -69,12 +72,49 @@ func create(name string, cfg *common.Config) (js []jobs.Job, endpoints int, err
...
+				logp.Warn("STRING HERE")
+				stdoutResults = append(stdoutResults, s)
+			} else if r, ok := line.(Results); ok {
+				status := "down"
+				rawResults = r.Raw
+
+				if len(r.Journeys) > 0 {
+					journey := r.Journeys[0]
+					if journey.error != nil {
+						status = "down"
```
Comment (vigneshshanmugam) [inline heartbeat/monitors/active/script/script.go:90]: should we do this from Node itself? feels cleaner? . each journey will have a status associated with it.

**Comment by wasserman** [comment]
Heartbeat Synthetics is a great idea!  I see it is just an experiment but I would look forward to such a feature.  I'm curious, what is driving the various browser or is it all simulated in Go?  I wouldn't mind playing with the experiment to try out some scenarios, especially if it could help the cause.  We are in the process of considering tools and products for Synthetics. I've done some limited scraping in Go and more sophisticated things in Python requests and Selenium.

**Comment by andrewvc** [comment]
@wasserman thanks for the interest! We'll be posting more about this soon, probably in the next month or so

Closing this issue in favor of https://github.com/elastic/beats/tree/heartbeat-synthetics

**Pull request closed without merging**